RFC-0003-arrow-flight-connector.md

```diff
@@ -0,0 +1,88 @@
+# **RFC0003 for Presto**
+
+## Arrow Flight connector template in Presto
+
+Proposers
+
+* Elbin Pallimalil
+
+## [Related Issues]
+
+https://github.com/prestodb/presto/issues/22729
+
+## Summary
+
+Create a base Arrow Flight module in Presto which can be extended to create an Arrow Flight connector.
+
+## Background
+
+Apache Arrow Flight is a new general-purpose client-server framework that simplifies high performance transport of large datasets over network interfaces. https://arrow.apache.org/blog/2019/10/13/introducing-arrow-flight/
+
+Presto currently supports different datasources through connectors for each type of datasource. The connectors can use JDBC or some other protocol for datasource connection. If we use Flight client to connect to the different datasources we can improve the performance of the datasource connection. Also we would only need to develop one Flight client that can be used across different datasources instead of developing individual connectors for each type of datasource.
+
+## Proposed Implementation
+
+A template for Flight connector will be created as `presto-base-arrow-flight` module. Since each implementation of flight server will be different, this base module can be extended to create a flight connector that supports a particular implementation. The template will support only read operations.
```

Comment:
This section is a bit confusing. What do you mean by template here ?

Reply:
This module will only provide the base classes needed to create a Flight connector. So this module is a template like `presto-base-jdbc` which can be extended to create a concrete implementation of Flight connector.

Reply:
Sounds good. Thanks !